go/mysql/conn.go

```diff
@@ -898,11 +898,8 @@ func (c *Conn) handleNextCommand(handler Handler) error {
 		if stmtID != uint32(0) {
 			defer func() {
 				prepare := c.PrepareData[stmtID]
-				if prepare.BindVars != nil {
-					for k := range prepare.BindVars {
-						prepare.BindVars[k] = nil
-					}
-				}
+				paramsCount := len(prepare.BindVars)
```

Comment:
We should just use `prepare.ParamsCount`. The value of `len(prepare.BindVars)` could actually be greater than `prepare.ParamsCount` when there are raw values in the query, which get converted into additional bindvars by the normalizer.

Reply:
Good call out.  Will adjust after the party tonight.  Thanks!

Reply:
Adjusted with https://github.com/vitessio/vitess/pull/5441/commits/0da79a88d17fc72be35731e6c002f41fee25b9d6.